0_portfolio_test.R

```diff
@@ -35,11 +35,19 @@ get_ald_scen <- function(portfolio_type) {
 
 get_ald_raw <- function(portfolio_type) {
   if (portfolio_type == "Equity") {
-    ald_raw <- read_rds(paste0(analysis_inputs_path, "/masterdata_ownership_datastore.rda"))
+    if (twodii_internal == FALSE & run_remotely == TRUE) {
```

Comment:
this makes me extremely nervous.... `get_ald_raw()` now has new dependencies on `twodii_internal` and `run_remotely` being available in the global environment, which makes debugging and stepping through the code more difficult, and could break something if `get_ald_raw()` ever gets called without it.

I would suggest...

Make `twodii_internal` and `run_remotely` (or one combined object/variable) arguments of the function `get_ald_raw()` and set sane defaults so that the default behaviour is always what it was before this change... then wherever you want to call this function and have a different behaviour (which I assume must be the case because otherwise you wouldn't be making this change), call this function and explicitly set this/these variable/s so that the function behaves the way you need it there. e.g.....

```r
get_ald_raw <- function(portfolio_type, stress_test_way == FALSE) {
  
  if (stress_test_way) {
    eq_filename <- "equity_ald_scenario_map.rda"
    bonds_filename <- "bonds_ald_scenario_map.rda"
  } else {
    eq_filename <- "masterdata_ownership_datastore.rda"
    bonds_filename <- "masterdata_debt_datastore.rda"
  }

  if (portfolio_type == "Equity") {
    ald_raw <- read_rds(file.path(analysis_inputs_path, eq_filename))
  }

  if (portfolio_type == "Bonds") {
    ald_raw <- read_rds(file.path(analysis_inputs_path, bonds_filename))
  }
}
```

then wherever you want this new behaviour, use...
```r
get_ald_raw(portfolio_type, stress_test_way == TRUE)
```

Reply:
valid point, will take care of that

Reply:
I agree with @cjyetman that this adds an invisible dependency. A different strategy would be to define a checker function, and set some flags via envvars or options, but I would prefer explicit arguments when possible.

Reply:
implemented

Reply:
perfect! thanks for being flexible with that